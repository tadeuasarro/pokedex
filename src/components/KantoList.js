/* eslint-disable */
import { useSelector } from 'react-redux';

const KantoList = () => {
  const counter = useSelector(state => state).pokemon.results;

  return(
    <div>
      {
        counter.map(pokemon => {
          return(
            <div key={pokemon.name}>{pokemon.name}</div>
          );
        })
      }
    </div>
  )


};

export default KantoList;
