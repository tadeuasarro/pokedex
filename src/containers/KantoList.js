import { useSelector } from 'react-redux';
import Pokemon from '../components/Pokemon';

const KantoList = () => {
  const counter = useSelector(state => state).pokemon.results;

  return (
    <div>
      {
        counter.map(pokemon => (
          <Pokemon key={pokemon.name} pokemon={pokemon} />
        ))
      }
    </div>
  );
};

export default KantoList;
