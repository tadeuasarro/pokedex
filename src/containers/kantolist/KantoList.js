import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { resetDetail } from '../../actions/index';
import Filtered from '../filtered/Filtered';
import Navbar from '../../components/navbar/Navbar';
import Listing from '../../components/listing/Listing';
import Loading from '../../components/loading/Loading';
import indexPokemon from '../../api/index-pokemon';
import './kanto-list.css';

const KantoList = () => {
  const { pokemon, filter, detail } = useSelector(state => state);

  const dispatch = useDispatch();

  if (!pokemon.results.length !== 0) {
    useEffect(() => dispatch(indexPokemon()), []);
  }

  if (detail.results) {
    dispatch(resetDetail());
  }

  if (pokemon.pending || filter.pending) return <Loading />;

  if (filter.results) {
    return (
      <div>
        <Navbar />
        <Filtered />
        <Listing pokemonList={filter.results} filter />
      </div>
    );
  }

  return (
    <div>
      <Navbar />
      <Filtered />
      <Listing pokemonList={pokemon.results} />
    </div>
  );
};

export default KantoList;
