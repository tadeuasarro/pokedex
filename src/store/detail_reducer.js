import { FETCH_DETAIL_ERROR, FETCH_DETAIL_PENDING, FETCH_DETAIL_SUCCESS } from '../actions/index';

const detailInitial = {
  pending: false,
  results: false,
  error: null,
};

export const detailReducer = (state = detailInitial, action) => {
  switch (action.type) {
    case FETCH_DETAIL_PENDING:
      return {
        ...state,
        pending: true,
      };
    case FETCH_DETAIL_SUCCESS:
      return {
        ...state,
        pending: false,
        results: action.detail,
      };
    case FETCH_DETAIL_ERROR:
      return {
        ...state,
        pending: false,
        error: action.error,
      };
    default:
      return state;
  }
};

export const getDetail = state => state.detail;
export const getDetailPending = state => state.pending;
export const getDetailError = state => state.error;
