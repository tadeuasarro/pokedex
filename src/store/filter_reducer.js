import { FETCH_TYPES_ERROR, FETCH_TYPES_PENDING, FETCH_TYPES_SUCCESS } from '../actions/index';

const detailInitial = {
  pending: false,
  results: false,
  error: null,
};

export const filterReducer = (state = detailInitial, action) => {
  switch (action.type) {
    case FETCH_TYPES_PENDING:
      return {
        ...state,
        pending: true,
      };
    case FETCH_TYPES_SUCCESS:
      return {
        ...state,
        pending: false,
        results: action.types,
      };
    case FETCH_TYPES_ERROR:
      return {
        ...state,
        pending: false,
        error: action.error,
      };
    default:
      return state;
  }
};

export const getTypes = state => state.types;
export const getTypesPending = state => state.pending;
export const getTypesError = state => state.error;
