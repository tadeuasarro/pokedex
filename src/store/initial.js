const pokemonInitial = {
  pending: false,
  results: [],
  error: null,
  detail: false,
};

export default pokemonInitial;
