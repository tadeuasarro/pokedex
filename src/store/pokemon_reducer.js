import { FETCH_POKEMON_PENDING, FETCH_POKEMON_SUCCESS, FETCH_POKEMON_ERROR } from '../actions/index';

const pokemonInitial = {
  pending: false,
  results: [],
  error: null,
};

export const pokemonReducer = (state = pokemonInitial, action) => {
  switch (action.type) {
    case FETCH_POKEMON_PENDING:
      return {
        ...state,
        pending: true,
      };
    case FETCH_POKEMON_SUCCESS:
      return {
        ...state,
        pokemon: action.pokemon,
      };
    case FETCH_POKEMON_ERROR:
      return {
        ...state,
        pending: false,
        error: action.error,
      };
    default:
      return state;
  }
};

export const getPokemon = state => state.results;
export const getPokemonPending = state => state.pending;
export const getPokemonError = state => state.error;
