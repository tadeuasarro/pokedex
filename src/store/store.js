import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { pokemonReducer } from './pokemon_reducer';
import { detailReducer } from './detail_reducer';

const store = createStore(
  combineReducers({
    pokemon: pokemonReducer,
    detail: detailReducer,
  }),
  applyMiddleware(thunk),
);

export default store;
