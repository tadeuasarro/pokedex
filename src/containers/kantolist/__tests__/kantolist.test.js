import { render, screen, act } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store/store';
import KantoList from '../KantoList';

describe('Rendering component', () => {
  it('creates an App component', () => {
    act(() => {
      render(
        <Provider store={store}>
          <KantoList />
        </Provider>,
      );
    });
    // eslint-disable-next-line
    const element = screen.getByText('Loading...');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const comp = renderer.create(
      <Provider store={store}>
        <KantoList />
      </Provider>,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
