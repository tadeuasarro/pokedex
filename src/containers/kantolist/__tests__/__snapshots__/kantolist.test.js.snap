// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div>
  Loading...
</div>
`;
